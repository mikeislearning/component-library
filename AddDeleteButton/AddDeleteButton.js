import React from 'react';
import { Image, TouchableOpacity } from 'react-native';
import { PropTypes } from 'prop-types';
import { images } from 'app/theme';

import styles from './AddDeleteButton.styles';

const TYPES = {
  add: 'add',
  delete: 'delete',
};

export default class AddDeleteButton extends React.Component {
  static propTypes = {
    accessibilityLabel: PropTypes.string,
    onPress: PropTypes.func.isRequired,
    type: PropTypes.oneOf([TYPES.add, TYPES.delete]),
  };

  render() {
    const { accessibilityLabel, onPress, type } = this.props;
    const containerStyle = [styles.container];
    const iconStyle = [styles.icon];
    if (type === TYPES.delete) {
      containerStyle.push(styles.containerSmall);
      iconStyle.push(styles.iconSmall);
    }

    return (
      <TouchableOpacity
        accessibilityLabel={accessibilityLabel}
        activeOpacity={0.7}
        onPress={onPress}
        style={containerStyle}
      >
        <Image source={images[type]} style={iconStyle} />
      </TouchableOpacity>
    );
  }
}
