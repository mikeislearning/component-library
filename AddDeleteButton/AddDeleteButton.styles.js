import { StyleSheet } from 'react-native';
import { colors } from 'app/theme';

const SIZE_SMALL = 21;
const SIZE_LARGE = 41;

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    borderColor: colors.blue,
    borderRadius: SIZE_LARGE,
    borderWidth: 1,
    height: SIZE_LARGE,
    justifyContent: 'center',
    width: SIZE_LARGE,
  },
  containerSmall: {
    height: SIZE_SMALL,
    width: SIZE_SMALL,
  },
  icon: {
    backgroundColor: colors.white,
    height: 20,
    width: 20,
  },
  iconSmall: {
    height: 12,
    width: 12,
  },
});

export default styles;
