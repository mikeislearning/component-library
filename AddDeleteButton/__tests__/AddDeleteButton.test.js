import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';

import AddDeleteButton from '../AddDeleteButton';

describe('<AddDeleteButton />', () => {
  it('renders the add button correctly', () => {
    const button = render(<AddDeleteButton onPress={() => {}} type="add" />);
    expect(button).toMatchSnapshot();
  });

  it('renders the delete button correctly', () => {
    const button = render(<AddDeleteButton onPress={() => {}} type="delete" />);
    expect(button).toMatchSnapshot();
  });

  it('can be pressed successfully', () => {
    const mockedFn = jest.fn();
    const { getByTestId } = render(
      <AddDeleteButton onPress={mockedFn} testID="addButton" type="add" />
    );
    fireEvent.press(getByTestId('addButton'));
    expect(mockedFn).toHaveBeenCalledTimes(1);
  });
});
