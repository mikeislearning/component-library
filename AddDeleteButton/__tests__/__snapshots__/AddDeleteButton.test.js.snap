// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddDeleteButton /> renders the add button correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#4145C8",
      "borderRadius": 41,
      "borderWidth": 1,
      "height": 41,
      "justifyContent": "center",
      "opacity": 1,
      "width": 41,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/add.png",
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "height": 20,
          "width": 20,
        },
      ]
    }
  />
</View>
`;

exports[`<AddDeleteButton /> renders the delete button correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#4145C8",
      "borderRadius": 41,
      "borderWidth": 1,
      "height": 21,
      "justifyContent": "center",
      "opacity": 1,
      "width": 21,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../app/images/delete.png",
      }
    }
    style={
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "height": 20,
          "width": 20,
        },
        Object {
          "height": 12,
          "width": 12,
        },
      ]
    }
  />
</View>
`;
